### PR TITLE
stellar-core 9.1.0 (new formula)

### PR DESCRIPTION
**Pull request by superstructor**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/stellar-core.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  desc "The backbone of the Stellar (XLM) network"
+  homepage "https://www.stellar.org/"
+  url "https://github.com/stellar/stellar-core.git",
+    :using => :git,
+    :tag => "v9.1.0",
+    :revision => "a278e95978bdac6d1015d82f4859dad780e752d3"
+  head "https://github.com/stellar/stellar-core.git", :using => :git
+
+  depends_on "pkg-config" => :run
+  depends_on "libsodium" => :run
```
Comment (fxcoudert) [inline Formula/stellar-core.rb]: Remove `:run` for `libsodium`, it is by default

File: Formula/stellar-core.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  depends_on "libtool" => :run
```
Comment (fxcoudert) [inline Formula/stellar-core.rb]: Are you sure `pkg-config` and `libtool` are really required at runtime?

File: Formula/stellar-core.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "postgresql" => :recommended
```
Comment (fxcoudert) [inline Formula/stellar-core.rb]: Let's make this mandatory, by removing `=> :recommended`. Fewer options are better for our users, who benefit from prebuilt binaries.

File: Formula/stellar-core.rb [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+      --disable-debug
+      --disable-dependency-tracking
+      --disable-silent-rules
+    ]
+    if build.with? "postgresql"
+      args << "--enable-postgres"
+    else
+      args << "--disable-postgres"
+    end
+    system "./configure", *args
```
Comment (fxcoudert) [inline Formula/stellar-core.rb]: After option is removed, inline all arguments after `system "./configure"` with no need for the `args` array

**Review by fxcoudert — CHANGES_REQUESTED** [review]
Thanks for the pull request!

File: Formula/stellar-core.rb [hunk 5]
```diff
@@ -0,0 +1,37 @@
+class StellarCore < Formula
+  desc "The backbone of the Stellar (XLM) network"
+  homepage "https://www.stellar.org/"
+  url "https://github.com/stellar/stellar-core.git",
+    :using => :git,
```
Comment (DomT4) [inline Formula/stellar-core.rb]: The `:using => :git` here and below are both redundant.

File: Formula/stellar-core.rb (hunk 1 above)
Comment (superstructor) [reply Formula/stellar-core.rb]: Done, thanks.

File: Formula/stellar-core.rb (hunk 2 above)
Comment (superstructor) [reply Formula/stellar-core.rb]: Your correct, only required for build, thanks.

File: Formula/stellar-core.rb (hunk 3 above)
Comment (superstructor) [reply Formula/stellar-core.rb]: Done, thanks.

File: Formula/stellar-core.rb (hunk 4 above)
Comment (superstructor) [reply Formula/stellar-core.rb]: Done, thanks.

File: Formula/stellar-core.rb (hunk 5 above)
Comment (superstructor) [reply Formula/stellar-core.rb]: Done, thanks.

**Comment by superstructor** [comment]
Thank you so much for the rapid response and helpful feedback! I've made the requested changes and retested.

File: Formula/stellar-core.rb [hunk 6]
```diff
@@ -0,0 +1,29 @@
+class StellarCore < Formula
+  desc "The backbone of the Stellar (XLM) network"
+  homepage "https://www.stellar.org/"
+  url "https://github.com/stellar/stellar-core.git",
```
Comment (fxcoudert) [inline Formula/stellar-core.rb]: We would prefer to build from a tarball (`https://github.com/stellar/stellar-core/archive/v9.1.0.tar.gz`) as this makes full use of Homebrew's caching mechanism.

File: Formula/stellar-core.rb [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+    :tag => "v9.1.0",
+    :revision => "a278e95978bdac6d1015d82f4859dad780e752d3"
+  head "https://github.com/stellar/stellar-core.git"
+
+  depends_on "libsodium"
+  depends_on "postgresql"
+  depends_on "pkg-config" => :build
```
Comment (fxcoudert) [inline Formula/stellar-core.rb]: Build deps first, ordered alphabetically among them. Then runtime deps.

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/stellar-core.rb (hunk 6 above)
Comment (DomT4) [reply Formula/stellar-core.rb]: Looks like there's submodules, which might not be part of the standard `archive` tarball? Can't remember off the top of my head 😓.

Comment (DomT4) [reply Formula/stellar-core.rb]: No, those folders are empty in the archive tarball. So would either need the `git clone` or a fuller release tarball, I guess.

File: Formula/stellar-core.rb (hunk 7 above)
Comment (superstructor) [reply Formula/stellar-core.rb]: Done, thanks.

File: Formula/stellar-core.rb (hunk 6 above)
Comment (superstructor) [reply Formula/stellar-core.rb]: Yes unfortunately doesn't appear possible to build from tarball at the moment. That is what I had attempted first. For the moment, this works and is not the only formula resorting to git. I'll also do a follow up pull request to improve this as soon as the situation changes.

**Comment by fxcoudert** [comment]
Thanks @superstructor for the pull request, and congratulations on your first Homebrew contribution!

**Pull request closed without merging**